pandas/io/sql.py

```diff
@@ -2024,6 +2042,255 @@ def _create_sql_schema(
 
 
 # ---- SQL without SQLAlchemy ---
+
+
+class ADBCDatabase(PandasSQL):
+    """
+    This class enables conversion between DataFrame and SQL databases
+    using ADBC to handle DataBase abstraction.
+
+    Parameters
+    ----------
+    con : adbc_driver_manager.dbapi.Connection
+    """
+
+    def __init__(self, con) -> None:
+        self.con = con
+
+    def execute(self, sql: str | Select | TextClause, params=None):
+        with self.con.cursor() as cur:
+            return cur.execute(sql)
+
+    def read_table(
+        self,
+        table_name: str,
+        index_col: str | list[str] | None = None,
+        coerce_float: bool = True,
+        parse_dates=None,
+        columns=None,
+        schema: str | None = None,
+        chunksize: int | None = None,
+        dtype_backend: DtypeBackend | Literal["numpy"] = "numpy",
+    ) -> DataFrame | Iterator[DataFrame]:
+        """
+        Read SQL database table into a DataFrame. Only keyword arguments used
+        are table_name and schema. The rest are silently discarded.
+
+        Parameters
+        ----------
+        table_name : str
+            Name of SQL table in database.
+        schema : string, default None
+            Name of SQL schema in database to read from
+
+        Returns
+        -------
+        DataFrame
+
+        See Also
+        --------
+        pandas.read_sql_table
+        SQLDatabase.read_query
+
+        """
+        if index_col:
+            raise NotImplementedError("'index_col' is not implemented for ADBC drivers")
+        if coerce_float is not True:
+            raise NotImplementedError(
+                "'coerce_float' is not implemented for ADBC drivers"
+            )
+        if parse_dates:
+            raise NotImplementedError(
+                "'parse_dates' is not implemented for ADBC drivers"
+            )
+        if columns:
+            raise NotImplementedError("'columns' is not implemented for ADBC drivers")
+        if chunksize:
+            raise NotImplementedError("'chunksize' is not implemented for ADBC drivers")
+
+        if schema:
+            stmt = f"SELECT * FROM {schema}.{table_name}"
+        else:
+            stmt = f"SELECT * FROM {table_name}"
+
+        mapping: type[ArrowDtype] | None | Callable
+        if dtype_backend == "pyarrow":
+            mapping = ArrowDtype
+        elif dtype_backend == "numpy_nullable":
+            from pandas.io._util import _arrow_dtype_mapping
+
+            mapping = _arrow_dtype_mapping().get
+        else:
+            mapping = None
+
+        with self.con.cursor() as cur:
+            return cur(stmt).fetch_arrow_table().to_pandas(types_mapper=mapping)
```

Comment:
Ah, I think because you are basically only testing `to_sql` .. All the read tests seem to rely on additional fixtures that have a connection + iris table set up, and those are not added for adbc.

With a quick hacky patch I can get the tests to fail with the "'Cursor' object is not callable" error:

```diff
--- a/pandas/tests/io/test_sql.py
+++ b/pandas/tests/io/test_sql.py
@@ -146,7 +146,9 @@ def create_and_load_iris_sqlite3(conn: sqlite3.Connection, iris_file: Path):
         reader = csv.reader(csvfile)
         next(reader)
         stmt = "INSERT INTO iris VALUES(?, ?, ?, ?, ?)"
-        cur.executemany(stmt, reader)
+        cur.executemany(stmt, list(reader))
+    conn.commit()
+    cur.close()
 
 
 def create_and_load_iris(conn, iris_file: Path, dialect: str):
@@ -532,6 +534,23 @@ def sqlite_iris_conn(sqlite_iris_engine):
         yield conn
 
 
+@pytest.fixture
+def sqlite_iris_adbc_conn(iris_path):
+    if pa_version_under8p0:
+        pytest.skip("ADBC requires pyarrow >= 8.0.0")
+    pytest.importorskip("adbc_driver_sqlite")
+    from adbc_driver_sqlite import dbapi
+
+    with tm.ensure_clean() as name:
+        uri = f"file:{name}"
+        with dbapi.connect(uri) as conn:
+            create_and_load_iris_sqlite3(conn, iris_path)
+            
+            yield conn
+            with conn.cursor() as cur:
+                cur.execute("DROP TABLE IF EXISTS test_frame")
+
+
 @pytest.fixture
 def sqlite_buildin():
     with contextlib.closing(sqlite3.connect(":memory:")) as closing_conn:
@@ -566,6 +585,7 @@ sqlite_iris_connectable = [
     "sqlite_iris_engine",
     "sqlite_iris_conn",
     "sqlite_iris_str",
+    "sqlite_iris_adbc_conn",
 ]
 
 sqlalchemy_connectable = mysql_connectable + postgresql_connectable + sqlite_connectable
```

Reply:
Ah cool - nice find